aodncore/pipeline/steps/upload.py

```diff
@@ -120,6 +121,16 @@ def run(self, pipeline_files):
                                                 dest_path=getattr(pipeline_file, self.dest_path_attr)))
             pipeline_file.is_stored = True
 
+        for pipeline_file in undo_deletions:
+            self._logger.info("undoing upload '{uri}'".format(uri=self._get_absolute_dest_uri(pipeline_file)))
+            try:
+                self._delete_file(pipeline_file)
+            except Exception as e:
+                raise FileDeleteFailedError(
+                    "{e}: '{dest_path}'".format(e=format_exception(e),
+                                                dest_path=getattr(pipeline_file, self.dest_path_attr)))
+            pipeline_file.is_undone = True
```

Comment:
See other comment. Having the attribute here means that a HARVEST_ONLY file would never have this attribute, even after the harvest step ran it through the undo process.

is_undone should be split into two attributes, something like is_storage_undone and is_harvest_undone set at the end of the run_undo_deletions method after each step has been confirmed to have been completed.